rancher2/resource_rancher2_app_v2.go

```diff
@@ -92,7 +98,7 @@ func resourceRancher2AppV2Read(d *schema.ResourceData, meta interface{}) error {
 			d.Set("cluster_name", cluster.Name)
 		}
 		if systemDefaultRegistry, ok := d.Get("system_default_registry").(string); !ok || len(systemDefaultRegistry) == 0 {
-			systemDefaultRegistry, err := meta.(*Config).GetSettingV2ByID(appV2DefaultRegistryID)
+			systemDefaultRegistry, err := meta.(*Config).GetSettingV2ByID(clusterID, appV2DefaultRegistryID)
```

Comment:
Settings are not defined for downstream clusters I guess. Correct me if I am wrong. 

Settings are always defined in the local cluster *only*. So, I think we need to either send "local" here or get back to the previous way where "local" is hardcoded.

Reply:
Done

Reply:
@rohitsakala In Rancher yeah, settings are only defined in the local cluster and a change to a settings field applies to all downstream clusters. Thanks for spotting this.